Update README.md

liamjwarren opened the pull request:
README.md updated with requirements for initial deployment submission. 

README.md now includes instructions on additional software required, building, and running the project.

liamjwarren committed:
Update README.md

colton-smith reviewed (changes requested):
I dont think an IDE is a requirement to run the engine or anything like that.

It would be better to include instructions specific to building with CMake, something like

```
# Create a build directory
mkdir build

# Execute cmake
cd build
cmake ..

# Run the generated executable
./aquarius
```

liamjwarren committed:
Update README.md

Changes to README.md to allow for building and running in a command-line interface instead on an IDE.

liamjwarren commented:
I have updated README.md to be now command-line interface based rather than IDE. I am not too familiar with CMake so if you know of a more efficient/right method than I did please let me know.

Burke-Daniel commented on README.md:
```diff
@@ -1,2 +1,21 @@
+## Building Aquarius
+1. Unzip the ZIP File
+2. Navigate in command-line interface to the unzipped project folder 
+3. Make a build directory by executing the following command `mkdir build`
+4. Navigate to the build directory by executing the following command `cd build`
+5. Execute CMake to build the project by executing the following command `cmake ..`
```
Do we want to include scripts at the root of the project (can have a .sh and a .bat so that we don't have a scripting language like python as a dependency) that does this bit for you?

Burke-Daniel commented on README.md:
```diff
@@ -1,2 +1,21 @@
+6. Generate an executable file by executing the following command `cmake --build . --config Release`
+
+## Running Aquarius
+1. Remain in the same command-line interface window as above, navigate to the Sandbox directory by executing 
+   the following command `cd Sandbox`    
+2. Run the generated executable by executing the following command `./Sandbox`
```
Is this `cd Sandbox` occurring within the build directory? Or the actual Sandbox directory of the project?
Is it `cd Aquarius/build/Sandbox && ./Sandbox` or `cd Aquarius/Sandbox && ./Sandbox` Just wanna make sure this is explicit.

liamjwarren replied:
`cd Aquarius/build/Sandbox && ./Sandbox` Good idea to state explicitly.

Burke-Daniel commented on README.md:
```diff
@@ -1,2 +1,21 @@
+3. Make a build directory by executing the following command `mkdir build`
+4. Navigate to the build directory by executing the following command `cd build`
+5. Execute CMake to build the project by executing the following command `cmake ..`
+6. Generate an executable file by executing the following command `cmake --build . --config Release`
```
I usually don't use the cmake command for building, but I wonder what this does on windows (I know on Linux and MacOS it just invokes make), either way it is nice that we do this in a platform-agnostic way (can't assume everyone is using make lol)

colton-smith replied:
Isnt make used in windows for building cmake projects?

Burke-Daniel replied:
ehhh it can be, but say you ran `cmake ..` with no args, it generates Visual Studio sln files. Like CLion comes with a version of make for windows so that's what it uses, and i'm pretty sure git for windows also comes with some sort of minGW make tool as well

colton-smith replied:
AH so when you invoke cmake --build it will use a platform specific build tool. In the case of windows, it uses MSBuild (same as UE4)

colton-smith commented on README.md:
```diff
@@ -1,2 +1,21 @@
+5. Execute CMake to build the project by executing the following command `cmake ..`
+6. Generate an executable file by executing the following command `cmake --build . --config Release`
+
+## Running Aquarius
```
Running Sample Aquarius Application*

colton-smith commented on README.md:
```diff
@@ -1,2 +1,21 @@
+6. Generate an executable file by executing the following command `cmake --build . --config Release`
+
+## Running Aquarius
+1. Remain in the same command-line interface window as above, navigate to the Sandbox directory by executing 
+   the following command `cd Sandbox`    
```
Navigate to build/Sandbox, this is a little confusing in case they close there console in between this section and the previous for some reason

colton-smith commented on README.md:
```diff
@@ -1,2 +1,21 @@
+1. Remain in the same command-line interface window as above, navigate to the Sandbox directory by executing 
+   the following command `cd Sandbox`    
+2. Run the generated executable by executing the following command `./Sandbox`
+3. A window will open, you have successfully built and run Aquarius!
```
**If** a window opens,

colton-smith commented on README.md:
```diff
@@ -1,2 +1,21 @@
+- CMake Version 3.10 ([Download](https://cmake.org/download/))
+
+## Building Aquarius
+1. Unzip the ZIP File
```
Extract the zip file to Aquarius/

Burke-Daniel reviewed (changes requested):
Couple small things for clarity, and lmk what you think of having build scripts and we can probably throw that in this PR too if we decide to do that. @colton-smith what do you think>

colton-smith commented on README.md:
```diff
@@ -1,2 +1,21 @@
+## Additional Required Software
+- CMake Version 3.10 ([Download](https://cmake.org/download/))
+
+## Building Aquarius
```
Mention that these commands and the application work for Windows, MAC, and Linux, but not WSL (windows subsystem for linux)

colton-smith reviewed (changes requested):
Couple of slight changes to make

Burke-Daniel commented:
Also not really important but @liamjwarren I noticed both commits have the same message summary, did you mean to rebase there?

liamjwarren committed:
Modifications to README.md

- Changes to sentences to make them easier to understand.
- Addition of commands and application compatibility statement.
- Some reformatting to ensure line lengths do not go too long.
- Change to Running Aquarius Header.

liamjwarren commented:
> Also not really important but @liamjwarren I noticed both commits have the same message summary, did you mean to rebase there?

liamjwarren committed:
Addition of Automatic Sections

-Added in two sections for automatic building and running using the scripts that @colton-smith developed.
-A few minor changes to sentences to try and make them more clear.
-Modifications to existing section headers to reflect changes.

colton-smith reviewed (approved):
Looks good, approved - but consider the question below. 

Wondering if we should have even included instructions about unzipping. The readme will be inside the zip folder so they will have to unzip first anyways to get to the readme. Perhaps we assume they have the project unzipped?

liamjwarren committed:
Removed unzipping instructions

-Removed unnecessary instructions regarding unzipping project folder.

liamjwarren commented:
> Looks good, approved - but consider the question below.
> 
> Wondering if we should have even included instructions about unzipping. The readme will be inside the zip folder so they will have to unzip first anyways to get to the readme. Perhaps we assume they have the project unzipped?

Good point, I have removed these instructions.

colton-smith commented on README.md:
```diff
@@ -1,2 +1,40 @@
+5. Generate an executable file by executing the following command `cmake --build . --config Release`
+
+## Running a Sample Aquarius Application Manually
+1. Remain in the same command-line interface window used to build the project, navigate to the Sandbox directory of the 
+   build directory by executing the following command `cd Sandbox`. If you have exited out of the command-line interface
+   used to build the project please navigate to `Aquarius/build/Sandbox` before proceeding to the next step. 
```
This whole point can be simplified by just sayingby just saying:
```
Navigate to Aquarius/build/Sandbox in terminal
```

colton-smith replied:
And for Windows, navigate to Aquarius\build\Sandbox\Release

liamjwarren committed:
Edits to manual run sample application

-Modify manual instructions to take into account the difference between OS.
-Modify a sentence to make it shorter.
